src/pydash/strings.py

```diff
@@ -2052,14 +2130,42 @@ def words(text, pattern=None):
     .. versionchanged:: 3.2.0
         Improved matching for one character words.
     """
-    return reg_exp_js_match(text, pattern or RE_WORDS)
+    if pattern is None:
+        if has_unicode_word(text):
+            return reg_exp_js_match(text, RE_UNICODE_WORDS)
+        else:
+            return reg_exp_js_match(text, RE_ASCII_WORDS)
+
+    return reg_exp_js_match(text, pattern)
 
 
 #
 # Utility functions not a part of main API
 #
 
 
+def compounder(text):
+    """
+    Remove single quote before passing into words(), required by certain functions such as
+    kebab_case, camel_case, start_case etc.
+
+    to match Lodash-style outputs
+    """
+    # Reference: https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/camelCase.js#L25 # noqa
+    return words(re.sub("['\u2019]", "", to_string(text)))
```

Comment:
Can use `RS_APOS` instead of `"['\u2019]"`.

Reply:
I have replaced them with RS_APOS in commit [74627df](https://github.com/dgilland/pydash/pull/159/commits/74627dfc3043a2964c3346209dd26fed519df87e). 
Furthermore, I have discovered that the Lodash includes the deburr() function in compounder (I have included the link in compounder function). The usage is to replace latin-1 supplementary letters to basic latin letters such as ç to c in multiple Unicode languages. I have generated some input strings, feed them into Lodash version's functions such as Camel Case, Kebab Case, Snake Case and use the outputs as the test cases. All passed after I include the deburr.